code/Spikes/Parser/Parser.cpp

```diff
@@ -0,0 +1,231 @@
+
+String Parser::Logoutmanagement(String ReceivedString, int Orders, int Time)
+{
+  String StringToBeReturned= "SIGN_OUT";
+  int pos = ReceivedString.indexOf("%");
+  String ReceivedPassword = ReceivedString.substring(pos+1);
+  if (ReceivedPassword==Password)
+  {
+    Returnvalue = 2;
+  }
+  else
+  {
+    Returnvalue = 3;
+  }
+  return StringToBeReturned;
+}
+
+/***********************************************************************************/
+//Function Ordermanagement:
+//->checks if password and order-string are ok
+//->returns the string to be sent to the App
+//->updates the Returnvalue
+
+String Parser::Ordermanagement(String ReceivedString, int Orders, int Time)
+{
+  String StringToBeReturned;
+  String CommonAnswer_Part1 = "ORDER_RS%";
+  String CommonAnswer_Part2 =  String(Orders) + "%" + String(Time);
+  int pos1 = ReceivedString.indexOf("%");
+  String SecondPart = ReceivedString.substring(pos1+1);
+  int pos2 = SecondPart.indexOf("%");
+  String ReceivedPassword = SecondPart.substring(0,pos2);
+  String ThirdPart = SecondPart.substring(pos2+1);
+  if (ReceivedPassword==Password)
+  {
+    StringToBeReturned = CommonAnswer_Part1 + "SUCCESSFULL%" + CommonAnswer_Part2;
+    Returnvalue = 4;
+    bool StringOK = CheckString(ThirdPart);
+    if (StringOK)
+    {
+      StringToBeReturned = CommonAnswer_Part1 + "SUCCESSFULL%" + CommonAnswer_Part2;
+      Returnvalue=4;
+    }
+    else
+    {
+      StringToBeReturned = CommonAnswer_Part1 + "ORDER_WRONG%" + CommonAnswer_Part2;
+      Returnvalue = 5;
+      g_counter_timer++;
+    }
+  }
+  else
+  {
+    StringToBeReturned = CommonAnswer_Part1 + "PW_WRONG%" + CommonAnswer_Part2;
+    Returnvalue = 6;
+  }
+  return StringToBeReturned;
+}
+
+/*****************************************************************************************/
+//Function Broadcastmanagement:
+//->returns the value to be sent to the App
+//->updates the Returnvalue
+
+String Parser::Broadcastmanagement()
+{
+  String BroadcastAnswer = "BROADCAST_RS%SMART_MINIFAB";
+  Returnvalue = 7;
+  return BroadcastAnswer;
+}
+
+/****************************************************************************************/
+//Function RunParser:
+//->returns the integer value (Returnvalue), which controls the main state machine
+//->updates the string to be sent to the App
+
+int Parser::RunParser(String ReceivedString,int Orders, int RemainingTime)
+{
+  String ReceivedString_NoDelim = ReceivedString.substring(1,ReceivedString.length()-1); 
+  int DelimiterPosition = ReceivedString_NoDelim.indexOf("%");
+  String FirstWord = ReceivedString_NoDelim.substring(0,DelimiterPosition);
+  if(FirstWord=="SIGN_IN")
+    {
+      Serial.println("Anmeldung");
+      Answer = Loginmanagement(ReceivedString_NoDelim,0,0);
+    }
+    else 
+      {
+        if (FirstWord=="ORDER")
+        {
+          Answer = Ordermanagement(ReceivedString_NoDelim,0,0);
+          Answer = "[" + Answer + "]\n";
+        }
+        else
```

Comment:
Vielleicht ist es rein optisch. Vielleicht uebersehe ich etwas. Sollte es nicht eher Folgendes sein:
```
if (FirstWord=="ORDER")
{
}
else if (FirstWord=="SIGN_OUT")
{
}
else if (FirstWord == "BROADCAST")
{
}
else
{
}
```

Es aendert nichts an der Logik, macht aber schoener deutlich, dass in allen Faellen das `FirstWord==XXXX` abgefragt wird im Gegensatz zu
```
if (FirstWord=="ORDER")
{
}
else
{
      if (FirstWord=="SIGN_OUT")
      {
      }
      else
      if (FirstWord == "BROADCAST")
      {
      }
      else
      {
      }
}
```

Edit: Ninjaedit for formatting.. :cat: